internal/search/query/parser.go

```diff
@@ -86,18 +88,25 @@ const (
 	RPAREN keyword = ")"
 )
 
-func isSpace(c byte) bool {
-	return (c == ' ') || (c == '\n') || (c == '\r') || (c == '\t')
+func isSpace(buf []byte) bool {
+	r, _ := utf8.DecodeRune(buf)
+	return unicode.IsSpace(r)
 }
 
-// skipSpace returns the number of spaces skipped from the beginning of a buffer buf.
+// skipSpace returns the number of spaces, corresponding to runes, skipped from the beginning of a buffer buf.
```

Comment:
This is a bit ambigious to me. Are you saying you return the number of whitespace runes, or the number of bytes? From the implementation (and from what seems more useful) I suppose you mean bytes?

```suggestion
// skipSpace returns the number of whitespace bytes skipped from the beginning of a buffer buf.
```